drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -244,12 +245,21 @@ func (k *k8sSchedOps) validateVolumeClone(parent *api.Volume, parentAnnotation s
 }
 
 func (k *k8sSchedOps) validateStorkClone(parent *api.Volume, snapshotAnnotation string) error {
+	volumeLabels := parent.Locator.VolumeLabels
+	if volumeLabels != nil {
+		snapName, ok := volumeLabels[storkSnapshotNameKey]
+		if ok && snapName == snapshotAnnotation {
+			return nil
+		}
+	}
+
 	parentName := parent.Locator.Name
-	if parentName != snapshotAnnotation {
-		return fmt.Errorf("Parent name [%s] does not match the snapshot annotation "+
-			"[%s] on the clone PVC", parentName, snapshotAnnotation)
+	if parentName == snapshotAnnotation {
```

Comment:
This will never match, right?

Reply:
I think old stork version will match this. That's why this check was present before.